### PR TITLE
[DEVOPS-1750] Fix go release

### PR DESCRIPTION
**Pull request by michalchecinski**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

- **.github/workflows/build-rust-cross-platform.yml:** Add ARM64 linux build
- **.github/workflows/release-go.yml** Download precompiled C artifacts and place them in the repo

## Before you submit

- Please add **unit tests** where it makes sense to do so

### CODE REVIEW

File: .github/workflows/build-rust-cross-platform.yml [hunk 1]
```diff
@@ -25,6 +26,8 @@ jobs:
             target: x86_64-pc-windows-msvc
           - os: ubuntu-22.04
             target: x86_64-unknown-linux-gnu
+          # - os: ubuntu-22.04
+          #   target: aarch64-unknown-linux-gnu
```
Comment (vgrassia) [inline .github/workflows/build-rust-cross-platform.yml:30]: Should this still be commented out?

File: .github/workflows/release-go.yml [hunk 2]
```diff
@@ -58,6 +58,75 @@ jobs:
         with:
           path: sdk
 
+      - name: Download artifacts
+        uses: bitwarden/gh-actions/download-artifacts@main
+        with:
+          workflow: generate_schemas.yml
+          path: sdk/languages/go/bitwarden_sdk_secrets/lib
+          workflow_conclusion: success
+          branch: ${{ github.event.inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```
Comment (vgrassia) [inline .github/workflows/release-go.yml:67]:
```suggestion
          branch: ${{ inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```

File: .github/workflows/release-go.yml [hunk 3]
```diff
@@ -58,6 +58,75 @@ jobs:
...
+          artifacts: schemas.go
+
+      - name: Download x86_64-apple-darwin artifact
+        uses: bitwarden/gh-actions/download-artifacts@main
+        with:
+          workflow: build-rust-cross-platform.yml
+          path: temp/macos-x64
+          workflow_conclusion: success
+          branch: ${{ github.event.inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```
Comment (vgrassia) [inline .github/workflows/release-go.yml:76]:
```suggestion
          branch: ${{ inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```

File: .github/workflows/release-go.yml [hunk 4]
```diff
@@ -58,6 +58,75 @@ jobs:
...
+          artifacts: libbitwarden_c_files-x86_64-apple-darwin
+
+      - name: Download aarch64-apple-darwin artifact
+        uses: bitwarden/gh-actions/download-artifacts@main
+        with:
+          workflow: build-rust-cross-platform.yml
+          workflow_conclusion: success
+          branch: ${{ github.event.inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```
Comment (vgrassia) [inline .github/workflows/release-go.yml:84]:
```suggestion
          branch: ${{ inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```

File: .github/workflows/release-go.yml [hunk 5]
```diff
@@ -58,6 +58,75 @@ jobs:
...
+          artifacts: libbitwarden_c_files-aarch64-apple-darwin
+          path: temp/macos-arm64
+
+      - name: Download x86_64-unknown-linux-gnu artifact
+        uses: bitwarden/gh-actions/download-artifacts@main
+        with:
+          workflow: build-rust-cross-platform.yml
+          workflow_conclusion: success
+          branch: ${{ github.event.inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```
Comment (vgrassia) [inline .github/workflows/release-go.yml:93]:
```suggestion
          branch: ${{ inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```

File: .github/workflows/release-go.yml [hunk 6]
```diff
@@ -58,6 +58,75 @@ jobs:
...
+          path: temp/linux-x64
+
+      # - name: Download arm64-unknown-linux-gnu artifact
+      #   uses: bitwarden/gh-actions/download-artifacts@main
+      #   with:
+      #     workflow: build-rust-cross-platform.yml
+      #     workflow_conclusion: success
+      #     branch: ${{ github.event.inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
+      #     artifacts: libbitwarden_c_files-arm64-unknown-linux-gnu
+      #     path: temp/linux-arm64
```
Comment (vgrassia) [inline .github/workflows/release-go.yml:104]: Should this still be commented out?

File: .github/workflows/release-go.yml [hunk 7]
```diff
@@ -58,6 +58,75 @@ jobs:
...
+
+      - name: Download x86_64-pc-windows-msvc artifact
+        uses: bitwarden/gh-actions/download-artifacts@main
+        with:
+          workflow: build-rust-cross-platform.yml
+          workflow_conclusion: success
+          branch: ${{ github.event.inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```
Comment (vgrassia) [inline .github/workflows/release-go.yml:111]:
```suggestion
          branch: ${{ inputs.release_type == 'Dry Run' && 'main' ||  github.ref_name }}
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

File: .github/workflows/release-go.yml (hunk 3 above)
Comment (michalchecinski) [reply .github/workflows/release-go.yml]: Fixed.

File: .github/workflows/release-go.yml (hunk 7 above)
Comment (michalchecinski) [reply .github/workflows/release-go.yml]: Fixed.

File: .github/workflows/release-go.yml (hunk 4 above)
Comment (michalchecinski) [reply .github/workflows/release-go.yml]: Fixed.

**Review by vgrassia — APPROVED** [review]

**Pull request merged**